### DISCUSSION TITLE
Dynamic table rows content

### DISCUSSION THREAD

**Discussion by JSR33** [QuestPDF/QuestPDF#328] [Q&A] [answered]
Hi.

I'm implementing a feature on my project where I need to fill my table vertically. Until here everything fine.
The problem is that I have dynamic content size and I don't know how many rows I'm able to create in one page (I don't want it to page break, I want it to continue in the next column).

Any ideas how I can control this?

EDIT: I saw that Dynamic Element was release on May and I am using an older version I'll try this new one :)

**Comment by bhaloditarun** [comment]
@JSR33 I have same issue: https://github.com/QuestPDF/QuestPDF/discussions/564

Do you find any solution for this? Please help, Thanks in advance!!

**Reply by JSR33** [reply]
@bhaloditarun sorry for late response. I'm now using DynamicComponents

**Reply by MarcinZiabek** [reply]
@JSR33 @bhaloditarun I must have missed your questions. I am sorry. 

Did you have a chance to try [the ShowEntire API](https://www.questpdf.com/api-reference/show-entire.html)? It ensures that its content does not get paged and is always visible as one block. 

[The EnsureSpace API](https://www.questpdf.com/api-reference/ensure-space.html) is a slightly safer approach that may prevent layout exceptions when your content takes more space than is available on a page, so it needs to be split to be properly rendered.

As far as I understand, it should cover the described scenario where content paging is not desired in the table, right? 😁

**Comment by MarcinZiabek** [comment, marked as answer]
I have just realized this workaround. Limitation: content in the item cannot be split between columns. The code could be nicely abstracted into a helper method:

```csharp
container
    .Padding(25)
    .FlipVertical()
    .RotateLeft()
    .Inlined(inlined =>
    {
        foreach (var i in Enumerable.Range(0, 50))
        {
            inlined
                .Item()
                .RotateRight()
                .FlipVertical()

                // your content:
                .MaxWidth(250)
                .Background(Placeholders.BackgroundColor())
                .Padding(10)
                .Text($"{i}: {Placeholders.Label()}")
                .FontSize(15);
        }
    });
```

That produces:
![image](https://github.com/QuestPDF/QuestPDF/assets/9263853/842a8edf-d7c8-46e8-8c87-38fce874a596)
![image](https://github.com/QuestPDF/QuestPDF/assets/9263853/16a626ff-de37-4493-aad2-c1997517bd31)
![image](https://github.com/QuestPDF/QuestPDF/assets/9263853/e5432831-9f8d-41da-b136-6680c1b32a65)

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
